### ISSUE TITLE
CRules.SendCommand sending twice with one call.

### ISSUE THREAD

**Issue by FuzzCat927** [transhumandesign/kag-base#267]
**Description**

CRules.SendCommand sending command twice.

**Reproduction**

Not sure. Mod files attached have debug statements to highlight this error. Can be found in Spell.as from the mod on line 289, 384, and 388.
[_Magic.zip](https://github.com/transhumandesign/kag-base/files/2590207/_Magic.zip)

1. Download mod.
2. Join sandbox gamemode
3. Press the r key.
4. Check server console, or just game console if on localhost.
5. The debug print that runs from the onCommand hook prints twice, while the debug prints surrounding the SendCommand only once.

**Expected Behavior**

The command should send once, not twice. 

**Screenshots**

[gif here](https://gyazo.com/7e1c8c0cdb73225d20a92e2d98179093)

**Setup**

- Win 10, but it would probably be worth checking if this happens on other OS. I don't have anything to test it on besides Win10.
- Build Number: 2760
- No hardware is modified or overclocked

**Comment by FuzzCat927** [comment]
Appears to be related to having the CBlob hooks and CRules hooks in the same file. Still worth looking at, though.

**Comment by makmoud98** [comment]
Me and FuzzCat figured out the issue. Apparently Spell.as was being used as a CBlob Script and a CRules script and for some reason that made the onCommand Hook call twice. Possibly needs some kind of engine-side check in order to stop modders from doing this or maybe allow it and fix the issues surrounding it.

**Comment by 1bardesign** [comment]
I don't believe this is the issue at all - oncommand was likely getting called on both server and client.
Commands can be sent from _anywhere_ for various tricky reasons, but most of the time they only **should** be sent from the owner player, or the server for non-player objects.

As such, this is not a bug in the engine so much as a bug in the mod in question - to fix it, just wrap the send in an `isServer` check.

**Comment by FuzzCat927** [comment]
It was already wrapped in a isClient check though

**Comment by 1bardesign** [comment]
There is not. I'll look into what's happening on localhost today.

However, it looks like `Spell.as` could be split into a rules and blob script (which both include `SpellCommon.as` with the spell class definition) to avoid having both hooks in one script if that indeed causes this issue.

**Comment by 1bardesign** [comment]
Debugged this today.

The issue is spell.as being included in another rules script as well as being used "standalone", so the hook is present twice and called twice.

The correct way to do this is to split it out the spell class definition to a separate file and never `#include` scripts with hooks in them.
